src/randoop/util/ListOfLists.java

```diff
@@ -95,4 +98,44 @@ public T get(int index) {
   public String toString() {
     return toJDKList().toString();
   }
+
+  @Override
+  public Iterator<T> iterator() {
+    return new LOLIterator(lists.iterator());
+  }
+  
+  private class LOLIterator implements Iterator<T> {
```

Comment:
Is LOL "list of lists"?  Please document.

But, couldn't we reuse functionality from some library instead of redefining it here?
For instance, plume-lib has MergedIterator that does the same as this, and other libraries must too.  And they won't contain the errors that I pointed out.